### PR TITLE
adds HA auth to the golang SDK

### PR DESCRIPTION
**Pull request by andrewpmartinez**
- adds message types for HA
- adds oidc auth, token refresh support
- adds oidc auth flows
- adds config support for multiple controllers

### CODE REVIEW

**Review by plorenz — APPROVED** [review]

File: ziti/client.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ import (
 	"fmt"
 	"github.com/openziti/foundation/v2/genext"
 	"github.com/openziti/transport/v2"
+	"github.com/pkg/errors"
```
Comment (ekoby) [inline ziti/client.go:30]: did we just remove it from main line?

File: ziti/ziti.go [hunk 2]
```diff
@@ -1099,6 +1101,49 @@ func (context *ContextImpl) getEdgeRouterConn(session *rest_model.SessionDetail,
...
+	for tuple := range context.routerConnections.IterBuffered() {
+		routerConn := tuple.Val
+		group.Add(1)
+
+		go func() {
+			err := routerConn.UpdateToken(newToken, timeout)
+
+			if err != nil {
+				if errs == nil {
+					errs = map[string]error{}
```
Comment (ekoby) [inline ziti/ziti.go:1120]: seem like a race condition

File: pb/edge_client_pb/edge_client.proto
Comment (ekoby) [inline pb/edge_client_pb/edge_client.proto:1]: what is the reason for protobuf? are you using it in edge communications?

File: ziti/contexts.go [hunk 3]
```diff
@@ -94,16 +95,52 @@ func NewContextWithOpts(cfg *Config, options *Options) (Context, error) {
...
+		ClientApiClient: edge_apis.NewClientApiClient(apiUrls[0], cfg.Credentials.GetCaPool(), func(codeCh chan string) {
+			provider := rest_model.MfaProvidersZiti
+
+			authQuery := &rest_model.AuthQueryDetail{
+				Provider: &provider,
+			}
+
+			newContext.Emit(EventAuthQuery, authQuery)
+
+			if *authQuery.Provider == rest_model.MfaProvidersZiti {
```
Comment (ekoby) [inline ziti/contexts.go:126]: is this always `true`?

**Pull request merged**